client/src/customHooks/useAuth.js

```diff
@@ -0,0 +1,65 @@
+import { useState, useEffect } from "react";
+import jwtDecode from "jwt-decode";
+import axios from "axios";
+import { useNavigate } from "react-router-dom";
+
+const useAuth = () => {
+  const navigate = useNavigate();
+  const [user, setUser] = useState(null);
+  const [clientId, setClientId] = useState("");
+
+  function handleSignUp() {
+    /* global google */
+    google.accounts.id.initialize({
+      client_id: clientId,
+      callback: handleCallbackResponse,
+    });
+    google.accounts.id.renderButton(document.getElementById("signInDiv"), {
+      theme: "outline",
+      size: "large",
+    });
+    google.accounts.id.prompt();
+  }
+
+  function handleCallbackResponse(response) {
+    if (response && response.credential) {
+      console.log("Encoded JWT ID token: " + response.credential);
+      let userObject = jwtDecode(response.credential);
+      console.log(userObject);
+      // Save the token in localStorage
```

Comment:
we don't need this comment as it is obvious. See this concept https://blog.codinghorror.com/code-tells-you-how-comments-tell-you-why/

Reply:
Fixed